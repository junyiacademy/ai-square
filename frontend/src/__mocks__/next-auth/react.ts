export const useSession = jest.fn(() => ({
  data: null,
  status: 'unauthenticated'
}));

export const signIn = jest.fn();
export const signOut = jest.fn();
export const getCsrfToken = jest.fn();
export const getSession = jest.fn();
export const getProviders = jest.fn();

export const SessionProvider = ({ children }: { children: React.ReactNode }) => children;