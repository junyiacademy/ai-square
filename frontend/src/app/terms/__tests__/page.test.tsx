import React from 'react';
import { render, screen } from '@testing-library/react';
import TermsOfServicePage from '../page';

// Mock next/link
jest.mock('next/link', () => {
  return ({ children, href }: { children: React.ReactNode; href: string }) => (
    <a href={href}>{children}</a>
  );
});

// Mock react-i18next
jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (key: string, defaultValue?: string) => defaultValue || key,
    i18n: { 
      changeLanguage: jest.fn(), 
      language: 'en' 
    }
  })
}));

describe('TermsOfServicePage', () => {
  it('should render without errors', () => {
    const { container } = render(<TermsOfServicePage />);
    expect(container).toBeTruthy();
  });

  it('should display the page title', () => {
    render(<TermsOfServicePage />);
    expect(screen.getByText('Terms of Service')).toBeInTheDocument();
  });

  it('should show effective date', () => {
    render(<TermsOfServicePage />);
    expect(screen.getByText(/Effective Date/)).toBeInTheDocument();
    expect(screen.getByText(/2025-07-01/)).toBeInTheDocument();
  });

  it('should show last updated date', () => {
    render(<TermsOfServicePage />);
    expect(screen.getByText(/Last Updated/)).toBeInTheDocument();
    expect(screen.getByText(/2025-06-30/)).toBeInTheDocument();
  });

  it('should display all major sections', () => {
    render(<TermsOfServicePage />);
    
    expect(screen.getByText('1. Acceptance of Terms')).toBeInTheDocument();
    expect(screen.getByText('2. Description of Service')).toBeInTheDocument();
    expect(screen.getByText('3. User Accounts')).toBeInTheDocument();
    expect(screen.getByText('4. Acceptable Use Policy')).toBeInTheDocument();
    expect(screen.getByText('5. Intellectual Property')).toBeInTheDocument();
    expect(screen.getByText('6. Privacy and Data Protection')).toBeInTheDocument();
    expect(screen.getByText('7. AI Services and Limitations')).toBeInTheDocument();
    expect(screen.getByText('8. Disclaimers and Limitations')).toBeInTheDocument();
    expect(screen.getByText('9. Modifications to Terms')).toBeInTheDocument();
    expect(screen.getByText('10. Termination')).toBeInTheDocument();
    expect(screen.getByText('11. Governing Law')).toBeInTheDocument();
    expect(screen.getByText('12. Contact Information')).toBeInTheDocument();
  });

  it('should display introduction text', () => {
    render(<TermsOfServicePage />);
    expect(screen.getByText(/Welcome to AI Square!/)).toBeInTheDocument();
  });

  it('should show service features', () => {
    render(<TermsOfServicePage />);
    expect(screen.getByText('AI literacy education content and assessments')).toBeInTheDocument();
    expect(screen.getByText('Problem-Based Learning (PBL) scenarios')).toBeInTheDocument();
    expect(screen.getByText('AI-powered tutoring and feedback')).toBeInTheDocument();
    expect(screen.getByText('Progress tracking and analytics')).toBeInTheDocument();
    expect(screen.getByText('Multi-language support (9 languages)')).toBeInTheDocument();
  });

  it('should show account types', () => {
    render(<TermsOfServicePage />);
    expect(screen.getByText('Student: Access to learning content and assessments')).toBeInTheDocument();
    expect(screen.getByText('Teacher: Additional class management features')).toBeInTheDocument();
    expect(screen.getByText('Administrator: Platform management capabilities')).toBeInTheDocument();
  });

  it('should show acceptable use policy', () => {
    render(<TermsOfServicePage />);
    expect(screen.getByText('Use the platform for educational purposes')).toBeInTheDocument();
    expect(screen.getByText('Misuse or attempt to manipulate the AI systems')).toBeInTheDocument();
  });

  it('should display contact email', () => {
    render(<TermsOfServicePage />);
    const emailLink = screen.getByRole('link', { name: /support@junyiacademy.org/i });
    expect(emailLink).toBeInTheDocument();
    expect(emailLink).toHaveAttribute('href', 'mailto:support@junyiacademy.org');
  });

  it('should have back to home link', () => {
    render(<TermsOfServicePage />);
    const homeLink = screen.getByText('Back to Home');
    expect(homeLink).toBeInTheDocument();
    expect(homeLink.closest('a')).toHaveAttribute('href', '/');
  });

  it('should display agreement notice', () => {
    render(<TermsOfServicePage />);
    expect(screen.getByText(/By using AI Square, you acknowledge that you have read and agree to these Terms of Service/)).toBeInTheDocument();
  });

  it('should have proper styling classes', () => {
    const { container } = render(<TermsOfServicePage />);
    const mainDiv = container.firstElementChild;
    expect(mainDiv).toHaveClass('min-h-screen');
    expect(mainDiv).toHaveClass('bg-gradient-to-br');
  });

  it('should render all sections with proper structure', () => {
    const { container } = render(<TermsOfServicePage />);
    const sections = container.querySelectorAll('section');
    expect(sections.length).toBeGreaterThan(10);
  });
});