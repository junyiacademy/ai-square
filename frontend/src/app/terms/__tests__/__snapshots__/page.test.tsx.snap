// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`page should match snapshot 1`] = `
<div
  class="min-h-screen bg-gradient-to-br from-gray-50 to-gray-100 dark:from-gray-900 dark:to-gray-800"
>
  <div
    class="max-w-4xl mx-auto px-4 sm:px-6 lg:px-8 py-12"
  >
    <div
      class="bg-white dark:bg-gray-800 rounded-xl shadow-lg p-8"
    >
      <div
        class="mb-8"
      >
        <h1
          class="text-3xl font-bold text-gray-900 dark:text-white mb-4"
        >
          terms.title
        </h1>
        <p
          class="text-sm text-gray-600 dark:text-gray-400"
        >
          terms.effectiveDate
          : 
          2025-07-01
           | 
          terms.lastUpdated
          : 
          2025-06-30
        </p>
      </div>
      <div
        class="prose prose-lg dark:prose-invert max-w-none"
      >
        <section
          class="mb-8"
        >
          <p
            class="text-gray-700 dark:text-gray-300"
          >
            terms.intro
          </p>
        </section>
        <section
          class="mb-8"
        >
          <h2
            class="text-2xl font-semibold mb-4"
          >
            terms.acceptance.title
          </h2>
          <p
            class="mb-4"
          >
            terms.acceptance.content
          </p>
          <p>
            terms.acceptance.capacity
          </p>
        </section>
        <section
          class="mb-8"
        >
          <h2
            class="text-2xl font-semibold mb-4"
          >
            terms.service.title
          </h2>
          <p
            class="mb-4"
          >
            terms.service.intro
          </p>
          <ul
            class="list-disc pl-6 space-y-2"
          >
            <li>
              terms.service.literacy
            </li>
            <li>
              terms.service.pbl
            </li>
            <li>
              terms.service.ai
            </li>
            <li>
              terms.service.tracking
            </li>
            <li>
              terms.service.multi
            </li>
          </ul>
        </section>
        <section
          class="mb-8"
        >
          <h2
            class="text-2xl font-semibold mb-4"
          >
            terms.accounts.title
          </h2>
          <h3
            class="text-xl font-medium mb-3"
          >
            terms.accounts.registration
          </h3>
          <ul
            class="list-disc pl-6 mb-4 space-y-2"
          >
            <li>
              terms.accounts.accurate
            </li>
            <li>
              terms.accounts.maintain
            </li>
            <li>
              terms.accounts.secure
            </li>
            <li>
              terms.accounts.responsible
            </li>
          </ul>
          <h3
            class="text-xl font-medium mb-3"
          >
            terms.accounts.types
          </h3>
          <ul
            class="list-disc pl-6 space-y-2"
          >
            <li>
              terms.accounts.student
            </li>
            <li>
              terms.accounts.teacher
            </li>
            <li>
              terms.accounts.admin
            </li>
          </ul>
        </section>
        <section
          class="mb-8"
        >
          <h2
            class="text-2xl font-semibold mb-4"
          >
            terms.use.title
          </h2>
          <h3
            class="text-xl font-medium mb-3"
          >
            terms.use.permitted
          </h3>
          <ul
            class="list-disc pl-6 mb-4 space-y-2"
          >
            <li>
              terms.use.educational
            </li>
            <li>
              terms.use.share
            </li>
            <li>
              terms.use.feedback
            </li>
          </ul>
          <h3
            class="text-xl font-medium mb-3"
          >
            terms.use.prohibited
          </h3>
          <ul
            class="list-disc pl-6 space-y-2"
          >
            <li>
              terms.use.misuse
            </li>
            <li>
              terms.use.share_credentials
            </li>
            <li>
              terms.use.cheat
            </li>
            <li>
              terms.use.harm
            </li>
            <li>
              terms.use.illegal
            </li>
            <li>
              terms.use.scrape
            </li>
          </ul>
        </section>
        <section
          class="mb-8"
        >
          <h2
            class="text-2xl font-semibold mb-4"
          >
            terms.ip.title
          </h2>
          <h3
            class="text-xl font-medium mb-3"
          >
            terms.ip.platform
          </h3>
          <p
            class="mb-4"
          >
            terms.ip.ownership
          </p>
          <h3
            class="text-xl font-medium mb-3"
          >
            terms.ip.user
          </h3>
          <p
            class="mb-4"
          >
            terms.ip.retain
          </p>
          <h3
            class="text-xl font-medium mb-3"
          >
            terms.ip.open
          </h3>
          <p>
            terms.ip.git
          </p>
        </section>
        <section
          class="mb-8"
        >
          <h2
            class="text-2xl font-semibold mb-4"
          >
            terms.privacy.title
          </h2>
          <p>
            terms.privacy.content
          </p>
        </section>
        <section
          class="mb-8"
        >
          <h2
            class="text-2xl font-semibold mb-4"
          >
            terms.ai.title
          </h2>
          <p
            class="mb-4"
          >
            terms.ai.disclaimer
          </p>
          <ul
            class="list-disc pl-6 space-y-2"
          >
            <li>
              terms.ai.educational
            </li>
            <li>
              terms.ai.verify
            </li>
            <li>
              terms.ai.improve
            </li>
          </ul>
        </section>
        <section
          class="mb-8"
        >
          <h2
            class="text-2xl font-semibold mb-4"
          >
            terms.disclaimers.title
          </h2>
          <p
            class="mb-4 uppercase font-semibold"
          >
            terms.disclaimers.asis
          </p>
          <p
            class="mb-4"
          >
            terms.disclaimers.liability
          </p>
        </section>
        <section
          class="mb-8"
        >
          <h2
            class="text-2xl font-semibold mb-4"
          >
            terms.modifications.title
          </h2>
          <p>
            terms.modifications.content
          </p>
        </section>
        <section
          class="mb-8"
        >
          <h2
            class="text-2xl font-semibold mb-4"
          >
            terms.termination.title
          </h2>
          <p
            class="mb-4"
          >
            terms.termination.rights
          </p>
        </section>
        <section
          class="mb-8"
        >
          <h2
            class="text-2xl font-semibold mb-4"
          >
            terms.law.title
          </h2>
          <p>
            terms.law.content
          </p>
        </section>
        <section
          class="mb-8"
        >
          <h2
            class="text-2xl font-semibold mb-4"
          >
            terms.contact.title
          </h2>
          <p
            class="mb-4"
          >
            terms.contact.intro
          </p>
          <div
            class="bg-gray-100 dark:bg-gray-700 p-4 rounded-lg"
          >
            <p
              class="font-medium"
            >
              AI Square
            </p>
            <p>
              Email: 
              <a
                class="text-blue-600 hover:underline"
                href="mailto:support@junyiacademy.org"
              >
                support@junyiacademy.org
              </a>
            </p>
          </div>
        </section>
      </div>
      <div
        class="mt-8 p-6 bg-blue-50 dark:bg-blue-900/20 rounded-lg"
      >
        <p
          class="text-center text-gray-700 dark:text-gray-300 mb-4"
        >
          terms.agreement
        </p>
      </div>
      <div
        class="mt-12 pt-8 border-t border-gray-200 dark:border-gray-700"
      >
        <a
          class="inline-flex items-center text-blue-600 hover:text-blue-700 dark:text-blue-400 dark:hover:text-blue-300"
          href="/"
        >
          <svg
            class="w-4 h-4 mr-2"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 19l-7-7m0 0l7-7m-7 7h18"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          common.backToHome
        </a>
      </div>
    </div>
  </div>
</div>
`;
